packages/report-flaky-tests/lib/github-api.ts

```diff
@@ -84,6 +86,44 @@ class GitHubAPI {
 
 		return data.merge_base_commit.commit;
 	}
+
+	async updateIssue(
+		params: Omit<
+			Exclude<
+				Parameters< Octokit[ 'rest' ][ 'issues' ][ 'update' ] >[ '0' ],
+				undefined
+			>,
+			'repo' | 'owner'
+		> & {
+			issue_number: number;
+		}
```

Comment:
What's this about?

Reply:
What is the `& { issue_number }` for?

Why can't we have `owner`?

I think it would be clearer and more valuable to use the type that is accepted by `octokit.rest.issues.update` which is `RestEndpointMethodTypes["issues"]["update"]["parameters"]` instead of using the `Parameters<>` helper. 

Right now this type doesn't offer much value because it's just restating the code that is below in `updateIssue` but in a different way. It's a bit like when you see a comment like this:

```js
// Print foo.
print( foo );
```

Ideally the type is a way in which we can express what our expectations are about the function. A bit like a unit test. The compiler then aids us by ensuring that the code matches our expectations. 

```ts
type UpdateIssueParams = Omit<
		RestEndpointMethodTypes["issues"]["update"]["parameters"],
		'repo'
>;
async updateIssue( params: UpdateIssueParams ) {
}
```

Reply:
> Why can't we have `owner`?

Because that's passed in the function body automatically.

Cool, didn't know that `@octokit/types` is available! I'll use it as it clearly makes the types easier to follow.

My only opinion on this is that I don't think assigning the type to a different name necessarily makes it easier to read and follow. If I hover on the `updateIssue` function in another file, I don't really want to see an arbitrary `UpdateIssueParams` name in the type, but rather prefer the resolved type showing what exactly it accepts. There's a balance to make though, as we don't want verbose types to be resolved recursively either.

Let me update the types and see if it's easy enough to follow.

Reply:
Yeah that's fair enough. Mostly I just hate using lots of helper types like `Parameters<>`, `Omit<>`, etc. because it means that you really have to think when hovering over a function and seeing the type. I don't ever want to think! 😛